app/models/spree/tax_rate.rb

```diff
@@ -30,16 +32,32 @@ def self.match(order)
       return [] if order.distributor && !order.distributor.charges_sales_tax
       return [] unless order.tax_zone
 
-      all.select do |rate|
-        rate.zone == order.tax_zone || rate.zone.contains?(order.tax_zone) || rate.zone.default_tax
+      all.includes(zone: { zone_members: :zoneable }).load.select do |rate|
```

Comment:
I think this could be extracted to a method with a good name:

`all.includes(zone: { zone_members: :zoneable }).load`

all_tax_rates_with_members?